.github/scripts/veristat-compare.py

```diff
@@ -38,165 +38,212 @@
+
+        if self.changes:
+            return "There are changes in verification performance"
+
+        return "No changes in verification performance"
+
+    def get_results_summary(self, html: bool = False) -> str:
+        title = self.get_results_title()
+        if not self.table:
+            return f"# {title}\n"
+
+        template = TEXT_SUMMARY_TEMPLATE
+        table = format_table(headers=SUMMARY_HEADERS, rows=self.table)
+
+        if html:
+            template = HTML_SUMMARY_TEMPLATE
+            table = html_decorate(table)
+
+        return template.format(title=title, table=table)
+
+
+def get_state_diff(value: str) -> float:
+    matches = re.match(TOTAL_STATES_DIFF_REGEX, value)
+    if percentage_diff := matches.group("percentage_diff"):
+        return float(percentage_diff)
+
+    raise ValueError(
+        f"Invalid {VeristatFields.TOTAL_STATES_DIFF} field value: {value}"
+    )
+
+
+def parse_table(csv_file):
+    reader = csv.DictReader(csv_file)
+    assert reader.fieldnames == VeristatFields.headers()
+
     new_failures = False
     changes = False
     table = []
 
-    with open(csv_filename, newline='') as file:
-        reader = csv.reader(file)
-        headers = next(reader)
-        if headers != HEADERS:
-            raise Exception(f'Unexpected table header for {filename}: {headers}')
-
-        for v in reader:
-            add = False
-            verdict = v[VERDICT_NEW]
-            diff = compute_diff(v)
-
-            if v[VERDICT_OLD] != v[VERDICT_NEW]:
-                changes = True
-                add = True
-                verdict = f'{v[VERDICT_OLD]} -> {v[VERDICT_NEW]}'
-                if v[VERDICT_NEW] == 'failure':
-                    new_failures = True
-                    verdict += ' (!!)'
-
-            if abs(diff * 100) > TRESHOLD_PCT:
-                changes = True
-                add = True
-
-            if not add:
-                continue
-
-            diff_txt = '{:+.1f} %'.format(diff * 100)
-            table.append([v[FILE], v[PROG], verdict, diff_txt])
-
-    return VeristatInfo(table=table,
-                        changes=changes,
-                        new_failures=new_failures)
-
-def format_table(headers, rows, html_mode):
-    def decorate(val, width):
-        s = str(val)
-        if html_mode:
-            s = s.replace(' -> ', ' &rarr; ');
-            s = s.replace(' (!!)', ' :bangbang: ');
-        return s.ljust(width)
-
-    column_widths = list(reduce(lambda acc, row: map(max, map(len, row), acc),
-                                rows,
-                                map(len, headers)))
-
-    with io.StringIO() as out:
-        def print_row(row):
-            out.write('| ')
-            out.write(' | '.join(map(decorate, row, column_widths)))
-            out.write(' |\n')
-
-        print_row(headers)
+    for record in reader:
+        add = False
+
+        verdict_old, verdict_new = (
+            record[VeristatFields.VERDICT_OLD],
```

Comment:
should you use `.value`?

```
>>> VeristatFields.FILE_NAME.value
'file_name'
```

Reply:
Yes, if you want to print it. But we add these into f-strings which calls `__str__()`. `VeristatFields(str, Enum)`, so `__str__()` for it will return `str` value of the enum by default